assets/js/components/SettingsNotice/SettingsNotice.js

```diff
@@ -33,12 +34,27 @@ import {
 	TYPE_SUGGESTION,
 	getIconFromType,
 } from './utils';
+import { ADSENSE_GA4_TOP_EARNING_PAGES_NOTICE_DISMISSED_ITEM_KEY as DISMISSED_KEY } from '../../modules/adsense/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { Button } from 'googlesitekit-components';
+
+const { useSelect, useDispatch } = Data;
 
 export default function SettingsNotice( props ) {
-	const { children, type, Icon = getIconFromType( type ) } = props;
+	const { children, type, dismiss, Icon = getIconFromType( type ) } = props;
+
+	const { dismissItem } = useDispatch( CORE_USER );
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed( DISMISSED_KEY )
```

Comment:
If the `dismiss` property is not empty then it contains the slug that we need to check for dismissal. So, you need to pass the `dismiss` property to this selector instead of the constants key.